common/cards/card-plugins/hermits/helsknight-rare.js

```diff
@@ -0,0 +1,91 @@
+import HermitCard from './_hermit-card'
+import {flipCoin, discardSingleUse} from '../../../../server/utils'
+import {GameModel} from '../../../../server/models/game-model'
+
+class HelsknightRareHermitCard extends HermitCard {
+	constructor() {
+		super({
+			id: 'helsknight_rare',
+			name: 'Helsknight',
+			rarity: 'rare',
+			hermitType: 'pvp',
+			health: 270,
+			primary: {
+				name: 'Pitfall',
+				cost: ['any'],
+				damage: 40,
+				power: null,
+			},
+			secondary: {
+				name: 'Trap Hole',
+				cost: ['pvp', 'pvp', 'pvp'],
+				damage: 100,
+				power:
+					'If opponent uses a single use effect card on their next turn, they must flip acoin. If heads, you take the card after its effect is applied and add it to your hand.',
+			},
+		})
+	}
+
+	/**
+	 * @param {GameModel} game
+	 * @param {string} instance
+	 * @param {import('../../../types/cards').CardPos} pos
+	 */
+	onAttach(game, instance, pos) {
+		const {player, otherPlayer} = pos
+
+		player.hooks.onAttack[instance] = (attack) => {
+			if (
+				attack.id !== this.getInstanceKey(instance) ||
+				attack.type !== 'secondary'
+			)
+				return
+
+			player.custom[instance] = true
+		}
+
+		otherPlayer.hooks.onAttack[instance] = () => {
+			if (!player.custom[instance]) return
+			otherPlayer.hooks.afterAttack[instance] = (attackResult) => {
+				if (attackResult.attack.type === 'effect') {
+					if (!otherPlayer.board.singleUseCard) return
+					const coinFlip = flipCoin(otherPlayer, this.id, 1)
+					otherPlayer.coinFlips[this.id] = coinFlip
+
+					if (coinFlip[0] == 'heads') {
+						player.hand.push(otherPlayer.board.singleUseCard)
```

Comment:
onDetach and the onDetach hooks should be called before removing the single use card, you can use moveCardToHand, I added it on #264.